### ISSUE TITLE
Found a BUG

### ISSUE THREAD

**Issue by zlel** [nicklasaven/TileLess#10]
interface.c

```
static RELATIONS *init_family(struct CTRL *parent)
{
    RELATIONS *t = malloc(sizeof(RELATIONS));
    if(!t)
    {
        log_this(100,"Failed to alloc memory in func %s",__func__);
        t->max_children = 0;
        return NULL;
    }
...
}
```

if 't' is null, then 't->max_children = 0;' will crash !

**Comment by zlel** [comment]
text.c

```
WCHAR_TEXT* init_wc_txt(size_t s)
{

    WCHAR_TEXT *t = malloc(sizeof(WCHAR_TEXT));
    if(!t)
    {
        log_this(100,"Failed to alloc memory in func %s",__func__);
        t->alloced = 0;
        return NULL;
    }
...
}
```

the same error with 't->alloced = 0'

**Comment by nicklasaven** [comment]
Did you get a crash? The code here just shows what happens if allocating memory fails. That is if you go out of ram. If it crashes the program I have failed to take care of that situation properly. That is possible. There is a lot of error cases that propably needs more attention. But I am not sure that is what you are pointing at here.

**Comment by zlel** [comment]
No. I just suspect that it will crash. 
malloc will return the null pointer when it fails. "t" pointer will be a null pointer. "t->max_children = 0;" program will crash after this statement refers to a null pointer.

**Comment by nicklasaven** [comment]
:-) thanks, you are right. Will fix when I get back to a computor.

**Comment by nicklasaven** [comment]
fixed with dec4edd038009bb6360fa3252f181df142f22b4d
